benchmarks/test_latency.py

```diff
@@ -0,0 +1,136 @@
+from megatron import get_args
+from megatron import get_timers
+from megatron import print_rank_0
+from megatron.checkpointing import load_checkpoint
+from megatron.initialize import initialize_megatron
+from megatron.model import GPTModel
+from megatron.training import get_model
+from megatron.arguments import core_transformer_config_from_args
+from megatron.text_generation import generate_and_post_process
+
+
+def model_provider(pre_process=True, post_process=True):
+    """Build the model."""
+
+    config = core_transformer_config_from_args(get_args())
+
+    print_rank_0('building GPT model ...')
+    model = GPTModel(
+        config,
+        num_tokentypes=0,
+        parallel_output=False,
+        pre_process=pre_process,
+        post_process=post_process
+    )
+
+    return model
+
+def add_text_generate_args(parser):
+    group = parser.add_argument_group(title='Benchmark latency for a single batch request')
+
+    group.add_argument("--num-iters", type=int, default=1,
+                       help='Number of requests to process.')
+    group.add_argument("--temperature", type=float, default=1.0,
+                       help='Sampling temperature.')
+    group.add_argument("--top_p", type=float, default=0.0,
+                       help='Top p sampling.')
+    group.add_argument("--top_k", type=int, default=0,
+                       help='Top k sampling.')
+    group.add_argument("--prompt-len", type=int, default=32,
+                       help='Length of prompt for each request.')
+    group.add_argument("--generate-len", type=int, default=1024,
+                       help='Length of generated text for each request.')
+    # group.add_argument("--dataset-path", type=str, default=None,
+    #                    help='Path to the requests data.')
+
+    return parser
+
+if __name__=="__main__":
+    initialize_megatron(extra_args_provider=add_text_generate_args,
+                        args_defaults={'tokenizer_type': 'AquilaTokenizer',
+                                       'no_load_rng': True,
+                                       'no_load_optim': True})
+    args = get_args()
+    timers = get_timers()
+    print(f"args is {args}")
+    if args.num_layers_per_virtual_pipeline_stage is not None:
+        print("Interleaved pipeline schedule is not yet supported for text generation.")
+        exit()
+    print_rank_0("WARNING: Forcing exit_on_missing_checkpoint to True for text "
+                 "generation.")
+    args.exit_on_missing_checkpoint = True
+
+    # Set up model and load checkpoint
+    model = get_model(model_provider, wrap_with_ddp=False)
+    print(f"get model successfully....")
+
+    if args.load is not None:
+        _ = load_checkpoint(model, None, None)
+    print("loading model successfully....")
+
+    assert len(model) == 1, "Above condition should have caught this"
+    model = model[0]
+
+    tokenizer = get_tokenizer()
+
+    prompt = ["我" * (args.prompt_len - 1)] * args.micro_batch_size
```

Comment:
Please add a path to a prompt file, which can include lots of prompts.

Reply:
Done.